int32 channel
float64 value
---
